src/firewall/core/fw_zone.py

```diff
@@ -329,18 +329,9 @@ def get_config_with_settings(self, zone):
         :return: exported config updated with runtime settings
         """
         obj = self.get_zone(zone)
-        conf_dict = self.get_config_with_settings_dict(zone)
-        conf_list = []
-        for i in range(16):  # tuple based API has 16 elements
-            if obj.IMPORT_EXPORT_STRUCTURE[i][0] not in conf_dict:
-                # old API needs the empty elements as well. Grab it from the
-                # class otherwise we don't know the type.
-                conf_list.append(
-                    copy.deepcopy(getattr(obj, obj.IMPORT_EXPORT_STRUCTURE[i][0]))
-                )
-            else:
-                conf_list.append(conf_dict[obj.IMPORT_EXPORT_STRUCTURE[i][0]])
-        return tuple(conf_list)
+        return obj.export_config_tuple(
+            self.get_config_with_settings_dict(zone), length=16
```

Comment:
`self.get_config_with_settings_dict(zone)` is passed as positional argument 1 (ignoring self). But there are no positional arguments for `export_config_tuple()`; only named arguments. I'm surprised python or flake8 is not complaining about this.

We should pass via named `conf_dict`.

Reply:
the signature is
```python
def export_config_tuple(self, conf_dict=None, length=None):
```

In this case, it's perfectly fine in Python to use positional arguments instead of named arguments. So it works just fine. Granted, for clarity, I often like to pass named arguments. I can do that here.

If it were
```python
def export_config_tuple(self, *, conf_dict=None, length=None):
```
then the `*` would enforce that the additional arguments are passed only as named arguments.

Will change.